### ISSUE TITLE
 InvalidParameterValue: AllocationStrategy must be "lowest-price", "capacity-optimized", or "diversified"

### ISSUE THREAD

**Issue by rwd5213** [aws/karpenter#1088]
## Version

**Karpenter**: v0.5.3

**Kubernetes**: v1.18.14

## Expected Behavior

Nodes are spun up for non-schedulable pods

## Actual Behavior

Karpenter throws an error 

```
ERROR controller.provissioning could not launch node, launching instances, creating fleet InvalidParameterValue: AllocationStrategy must be "lowest-price", "capacity-optimized", or "diversified"
```

## Steps to Reproduce the Problem

The error is difficult to reproduce since this error is appears in an AWS cloud environment with no Spot instances so I am guessing the the EC2 Fleet API is not familiar with the CapacityOptimizedPrioritized Allocation Strategy that Karpenter defaults to. 

Even when restricting Karpenter to only use on-demand instance capacity, I still see this error which is strange since it seems like lowest-price is what defined on line 131 of pkg/cloudprovider/aws/instance.go for on demand.

## Resource Specs and Logs

n/a

**Comment by bwagner5** [comment]
Hey @rwd5213! Did this error come from the distributed images on ECR or is this recompiled locally? The reason I ask is that it seems the aws-sdk  used in the error you're reporting is old (PRE v1.37.26 [2021-03-08]) which is when `capacity-optimized-prioritized` was released.  

Also, what region are you seeing this error?

**Comment by rwd5213** [comment]
Oh thats super weird, I did not recompile it locally, i used the images from ECR. I can double check the images I downloaded and transferred but the image is `public.ecr.aws/karpenter/controller:v0.5.3@sha256:ddd24d756cb324cf8f91f2274621646f83d6121ed6856312ca672a5f78c57174` 

This is in the us-iso-east-1 region.

**Comment by bwagner5** [comment]
@rwd5213 the PR above should fix the issue you are seeing.  The issue is that SpotOptions are included in all CreateFleet requests but Spot is not necessarily available in all regions, hence the error you are seeing.

**Comment by bwagner5** [comment]
Patch has been merged! We will release a new version of Karpenter at the end of this week or early next week.

**Comment by bwagner5** [comment]
The fix has been released in v0.5.5

**Comment by rwd5213** [comment]
Thank you! tested it out today and worked great!
